python/grass/temporal/spatial_topology_dataset_connector.py

```diff
@@ -118,31 +118,31 @@ def get_number_of_spatial_relations(self):
         relations = {}
         try:
             relations["equivalent"] = len(self._spatial_topology["EQUIVALENT"])
-        except:
+        except (KeyError, TypeError):
```

Comment:
The class, e.g., `def append_equivalent(self, map):`, always creates the dict item with an empty list, so the `len` call should always succeed, so there should be no TypeError, or if there is, it is programmer's fault and traceback is the desired behavior.

Reply:
Keeping ```except Exception``` for now?

Reply:
I meant the opposite. If you analyze the code, it should reveal that of the dict item is there, it will be an empty list. So, no TypeError. The attribute seems to be present. (Please check.) This would leave KeyError as what the original author intended to catch here.